modules/backup-gcs/client.go

```diff
@@ -216,32 +182,32 @@ func (g *gcsClient) PutObject(ctx context.Context, backupID, key string, byes []
 	return nil
 }
 
-func (g *gcsClient) Initialize(ctx context.Context, backupID string) error {
+func (g *gcsClient) Initialize(ctx context.Context, backupID, bucketName, bucketPath string) error {
 	key := "access-check"
 
-	if err := g.PutObject(ctx, backupID, key, []byte("")); err != nil {
-		return errors.Wrap(err, "failed to access-check gcs backup module")
+	if err := g.PutObject(ctx, backupID, key, bucketName, bucketPath, []byte("")); err != nil {
+		return errors.Wrap(err, "failed to access-check gcs backup module"+bucketName+bucketPath+backupID+key)
 	}
 
-	bucket, err := g.findBucket(ctx)
+	bucket, err := g.findBucket(ctx, "")
 	if err != nil {
 		return errors.Wrap(err, "find bucket")
 	}
 
-	objectName := g.makeObjectName(backupID, key)
+	objectName := g.makeObjectName(bucketPath, []string{backupID, key})
 	if err := bucket.Object(objectName).Delete(ctx); err != nil {
-		return errors.Wrap(err, "failed to remove access-check gcs backup module")
+		return errors.Wrap(err, "failed to remove access-check gcs backup module"+objectName)
 	}
 
 	return nil
 }
 
 // WriteToFile downloads an object and store its content in destPath
 // The file destPath will be created if it doesn't exit
-func (g *gcsClient) WriteToFile(ctx context.Context, backupID, key, destPath string) (err error) {
-	bucket, err := g.findBucket(ctx)
+func (g *gcsClient) WriteToFile(ctx context.Context, backupID, key, destPath, bucketName, bucketPath string) (err error) {
+	bucket, err := g.findBucket(ctx, bucketName)
 	if err != nil {
-		return fmt.Errorf("find bucket: %w", err)
+		return fmt.Errorf("(WriteToFile) find bucket: '%w'", err)
```

Comment:
can we remove `(WriteToFile)` from the error message?

Reply:
can we have unique error messages?

Reply:
and in any case, the error message is wrong.  findbucket should report "findbucket: ...".  writetofile should report "writetofile: ...".

Reply:
yes, I agree on unique error messages, but it should be done in a separate PR

Reply:
fmt.Errorf("(read) find bucket: '%w'", err)